### PR TITLE
Improve file size formatting

### PR DESCRIPTION
**Pull request by vxgmichel**
Improve the file size formatting by displaying for significant figures when needed.

### CODE REVIEW

**Review by touilleMan — COMMENTED** [review]
what about displaying `0.9MB` format ? using `1023KB` seems weird
(the rule would be: if the value is > 900, use the bigger unit)

File: parsec/core/gui/file_size.py [hunk 1]
```diff
@@ -3,122 +3,45 @@
...
-    '2M'
-
+def size(bytes, system):
+    """
+    Format the specified number of bytes with the corresponding system.
+
+    More specifically:
+    - `0 <= bytes < 10`:         1 significant figures:    `X B`
+    - `10 <= bytes < 100`:       2 significant figures:   `XY B`
+    - `100 <= bytes < 100`:      3 significant figures:  `XYZ B`
```
[the remark this answers is not in the document]
Comment (vxgmichel) [reply parsec/core/gui/file_size.py:13]: :+1:

File: parsec/core/gui/file_size.py [hunk 2]
```diff
@@ -3,122 +3,45 @@
...
-            suffix = singular
-        else:
-            suffix = multiple
-    return f"{amount} {suffix}"
+    # Convert to the right unit
+    amount = bytes / factor
+    # Pick the right formatter
+    formatter = ".1f" if amount < 99.95 and factor > 1 else ".0f"
+    # Format the amount and add the suffix
+    formatted_amount = format(bytes / factor, formatter)
```
[the remark this answers is not in the document]
Comment (vxgmichel) [reply parsec/core/gui/file_size.py:29]: :+1:

**Comment by vxgmichel** [comment]
> what about displaying 0.9MB format ? using 1023KB seems weird
(the rule would be: if the value is > 900, use the bigger unit)

Good catch, I forgot about the 1000-1024 range.
Do you know how windows display those files in the explorer?

File: parsec/core/gui/file_size.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (vxgmichel) [reply parsec/core/gui/file_size.py]: a8e1422329df1e3b799e6fac65694876d8ba7769

File: parsec/core/gui/file_size.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (vxgmichel) [reply parsec/core/gui/file_size.py]: a8e1422329df1e3b799e6fac65694876d8ba7769

**Comment by touilleMan** [comment]
> Do you know how windows display those files in the explorer?
image

image

image

image
buuuut
image


So in the end Windows round per KB in the explorer (size <1024B => 1KB, 1024B<= size < 2048B ==> 2KB etc.) and use the weird 1023KB display in the property view, unless size is big enough where some magic kicks in and now 0.99MB is displayed :/

Bottom story: it is a mess, nobody really care about this mess anyway, we should do what is convenient for us

**Comment by vxgmichel** [comment]
@touilleMan 
> it is a mess, nobody really care about this mess anyway, we should do what is convenient for us

Alright, could you quickly review the following commit? 8c67484

It implements the following policy:

```
    - `0 <= bytes < 10`:          1 significant figures:     `X B`
    - `10 <= bytes < 100`:        2 significant figures:    `XY B`
    - `100 <= bytes < 1000`:      3 significant figures:   `XYZ B`
    - `1000 <= bytes < 1024`:     2 significant figures: `0.9X KB`
    - `1 <= kilobytes < 10`:      2 significant figures:  `X.Y KB`
    - `10 <= kilobytes < 100`:    3 significant figures: `XY.Z KB`
    - `100 <= kilobytes < 1000`:  3 significant figures:  `XYZ KB`
    - `1000 <= kilobytes < 1024`: 2 significant figures: `0.9X MB`
    - And so on for MB, GB and TB
```

**Review by touilleMan — APPROVED** [review]
lgtm

**Pull request merged**